reactor-core/src/test/java/reactor/core/publisher/FluxSwitchOnFirstTest.java

```diff
@@ -98,6 +101,71 @@ public void shouldNotSubscribeTwiceConditional() {
                   .hasMessage("FluxSwitchOnFirst allows only one Subscriber");
     }
 
+    @Test
+    public void shouldNotSubscribeTwiceWhenCanceled() {
+        Schedulers.single().start();
+        CountDownLatch latch = new CountDownLatch(1);
+        StepVerifier.create(Flux.just(1L)
+                                .doOnComplete(() -> {
+                                    try {
+                                        latch.await();
+                                    }
+                                    catch (InterruptedException e) {
+                                        throw new RuntimeException(e);
+                                    }
+                                })
+                                .hide()
+                                .publishOn(Schedulers.parallel())
+                                .cancelOn(NoOpsScheduler.INSTANCE)
+                                .doOnCancel(latch::countDown)
+                                .switchOnFirst((s, f) -> f)
+                                .doOnSubscribe(s ->
+                                    Schedulers
+                                        .single()
+                                        .schedulePeriodically(s::cancel, 10, 0, TimeUnit.MILLISECONDS)
+                                )
+        )
+                    .expectSubscription()
+                    .expectNext(1L)
+                    .expectNoEvent(Duration.ofMillis(200))
+                    .thenCancel()
+                    .verifyThenAssertThat()
+                    .hasNotDroppedErrors();
+    }
+
+    @Test
+    public void shouldNotSubscribeTwiceConditionalWhenCanceled() {
+        Schedulers.single().start();
+        CountDownLatch latch = new CountDownLatch(1);
+        StepVerifier.create(Flux.just(1L)
+                                .doOnComplete(() -> {
+                                    try {
+                                        latch.await();
+                                    }
+                                    catch (InterruptedException e) {
+                                        e.printStackTrace();
```

Comment:
other test re-throws it, I would do the same here

Reply:
(it's not possible to avoid the `try/catch` inside the lambda, checked exceptions for the win)

Reply:
`shouldNotSubscribeTwiceWhenCanceled` re-throws, and `shouldNotSubscribeTwiceConditionalWhenCanceled` uses `printStackTrace`, I call for consistency :)

Reply:
fixed